### PR TITLE
Add wrapper for generic operations #1

### PR DESCRIPTION
**Pull request by safrio**

### CODE REVIEW

File: app/services/chatbot_service.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+    # send message to basecamp
+    HTTParty.post(url, query: { content: message })
+  end
+
+  def confirm_subscribe!
+    return nil unless url.present?
+    puts url
+    HTTParty.get(url)
+  end
```
Comment (Fodoj) [inline app/services/chatbot_service.rb]: This looks like SNS-only method => it doesn't belong to Chatbot class

Comment (Fodoj) [reply app/services/chatbot_service.rb]: Also it should be "confirm_subscription". Why exclamation mark in the name?

File: app/services/chatbot_service.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+class ChatbotService
+  attr_reader :url
+
+  def initialize(url)
+    @url = url
+  end
+
+  def post_message(options)
+    message = "Subject: #{options[:subject]}<br/> #{options[:message]}"
```
Comment (Fodoj) [inline app/services/chatbot_service.rb]: Let's keep message formatting out of this class. Perhaps makes sense to add something like "MessageTempate" class

File: config/service.yml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 default: &default
-  basecampbot_url: https://3.basecamp.com/3601306/integrations/6Mc7Rhug2NkpDY53Qu2x7TGn/buckets/1925463/chats/294329172/lines
-  snstopic_arn: arn:aws:sns:eu-central-1:798348585423:basecamp-bot
+  basecampbot_url: https://3.basecamp.com/3648568/integrations/jSQBpBexDzgJ6sTUSEnT9foQ/buckets/2496221/chats/362174890/lines
+  snstopic_arn: arn:aws:sns:us-west-2:482116491549:test_t
```
Comment (Fodoj) [inline config/service.yml]: I guess this should not be present in github repo

File: app/controllers/integrations/amazon_basecamp_controller.rb [hunk 4]
```diff
@@ -1,11 +1,7 @@
-require 'json'
-require 'httparty'
-
-class Api::V1::MessagesController < ApplicationController
+class Integrations::AmazonBasecampController < ApplicationController
```
Comment (Fodoj) [inline app/controllers/integrations/amazon_basecamp_controller.rb]: this is chatbot for basecamp, so having Basecampe in title of every controller makes little sense. "AwsSnsController" would be better, as it's really only for SNS.

File: app/controllers/integrations/amazon_basecamp_controller.rb [hunk 5]
```diff
@@ -25,10 +21,10 @@ def create
     end
 
     if amz_message_type.to_s.downcase == 'notification'
-      # DO WORK HERE
-      message = "Subject: #{request_body['Subject']}<br/> #{request_body['Message']}"
-      # send message to basecamp
-      HTTParty.post basecampbot_url, query: { content: message }
+      ChatbotService.new(Rails.configuration.service['basecampbot_url']).post_message(
```
Comment (Fodoj) [inline app/controllers/integrations/amazon_basecamp_controller.rb]: 1. Why create an instnace with `new` when it is more or less stateless?
2. Why have `service` in the name of the class?

File: config/service.yml [hunk 6]
```diff
@@ -1,7 +1,7 @@
 default: &default
-  basecampbot_url: https://3.basecamp.com/3648568/integrations/jSQBpBexDzgJ6sTUSEnT9foQ/buckets/2496221/chats/362174890/lines
-  snstopic_arn: arn:aws:sns:us-west-2:482116491549:test_t
-
+  basecampbot_url: https://3.basecamp.com/3601306/integrations/6Mc7Rhug2NkpDY53Qu2x7TGn/buckets/1925463/chats/294329172/lines
+  snstopic_arn: arn:aws:sns:eu-central-1:798348585423:basecamp-bot
+
```
Comment (Fodoj) [inline config/service.yml]: this whole file should be removed from git

File: app/services/chatbot/message_tempate.rb [hunk 7]
```diff
@@ -0,0 +1,5 @@
+module Chatbot::MessageTempate
+	def self.show(params)
+	  ActionView::Base.new(ActionController::Base.view_paths, message_hash: params).render(template: "/services/chatbot_message_tempate")
+	end
```
Comment (Fodoj) [inline app/services/chatbot/message_tempate.rb]: No tabs please, only two spaces for indentation

File: app/views/services/chatbot_message_tempate.html.erb [hunk 8]
```diff
@@ -0,0 +1,3 @@
+Subject: <%= @message_hash[1][:subject] %><br>
+Message:<br>
+<%= @message_hash[0] %>
```
Comment (Fodoj) [inline app/views/services/chatbot_message_tempate.html.erb]: Now let's find a way to have one template per integration type

Comment (Fodoj) [reply app/views/services/chatbot_message_tempate.html.erb]: Actually, forget about it. Not needed.

File: app/services/chatbot/message_tempate.rb [hunk 9]
```diff
@@ -0,0 +1,5 @@
+module Chatbot::MessageTempate
+	def self.show(params)
+	  ActionView::Base.new(ActionController::Base.view_paths, message_hash: params).render(template: "/services/chatbot_message_tempate")
```
Comment (Fodoj) [inline app/services/chatbot/message_tempate.rb]: why not:

```
ActionView::Base.new(ActionController::Base.view_paths, subject: subject, message: message)
```

Comment (safrio) [reply app/services/chatbot/message_tempate.rb]: Are you sure message template can't have more parametrs? In other services

Comment (Fodoj) [reply app/services/chatbot/message_tempate.rb]: I think we should try to standardize it. Subject, body, action links.

**Pull request closed without merging**